### PR TITLE
Adding an HLT filter for matching between PPS local tracks and central jet properties

### PR DESCRIPTION
**Pull request by maaraujo94**
#### PR description:

This PR implements an HLT filter which requires the matching between the proton track information from the PPS detector and the jet properties from the central CMS detector.

The filter can require matching of mass+rapidity and/or of xi - fractional energy loss of the scattered proton.

#### PR validation:

Filter has been run on data and MC and has worked as expected. More information on tests done is shown [here](https://indico.cern.ch/event/923848/contributions/3881644/attachments/2049163/3434200/0602_HLTforPPS.pdf.)

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

PR is not a backport.

cc: @fabferro @jjhollar @kshcheli @jan-kaspar

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
-1
Class name must be the same as file name and start with HLT... .
Header file not needed unless another cc file would need to include it.
Should use ```std::abs``` instead of ```abs```
This filter does not use/does not inherit from / any HLTFilter functionality, so why does it have to be an HLTFilter in an HLT package?

**Comment by maaraujo94** [comment]
The issues indicated have been corrected.
Regarding the last issue, I am not sure what is meant. We are using an EDFilter because the filter does not use HLTFilter functionality, following the advice given in a [previous PR](https://github.com/cms-sw/cmssw/pull/22446). We are submitting this filter as it is meant to be used in an HLT trigger proposal. If we should change from EDFilter to something else, of course we will be happy to change it.

**Comment by maaraujo94** [comment]
My apologies for the hasty commit that led to issues with compilation. I have now fixed the issue. 
I have had to add the directory "CondFormats/RunInfo" to the BuildFile xml, as the code requires LHCInfo.h, from that directory

File: HLTrigger/special/plugins/HLTPPSJetComparisonFilter.cc [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+      ->setComment("maximum relative deviation of RP m from dijet m- Used with do_my option");
+  desc.add<double>("maxDiffy", 1.)
+      ->setComment("maximum absolute deviation of RP y from dijet y. Used with do_my option");
+
+  desc.add<unsigned int>("nJets", 2)->setComment("number of jets to be used");
+
+  desc.add<bool>("do_xi", true)->setComment("flag to require xi matching");
+  desc.add<bool>("do_my", false)->setComment("flag to require m,y matching");
+
+  descriptions.add("hltCTPPSKinematicFilter", desc);
```
Comment (Martin-Grunewald) [inline HLTrigger/special/plugins/HLTPPSJetComparisonFilter.cc:84]: Should be ```descriptions.addWithDefaultLabel(desc);```

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**